model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -835,81 +835,82 @@ def _do_diffusion_step(
             "running stencils 07 08 09 10 (apply_diffusion_to_w_and_compute_horizontal_gradients_for_turbulence): end"
         )
 
-        log.debug(
-            "running fused stencils 11 12 (calculate_enhanced_diffusion_coefficients_for_grid_point_cold_pools): start"
-        )
-
-        self.calculate_enhanced_diffusion_coefficients_for_grid_point_cold_pools.with_connectivities(
-            self.compile_time_connectivities
-        )(
-            theta_v=prognostic_state.theta_v,
-            theta_ref_mc=self._metric_state.theta_ref_mc,
-            thresh_tdiff=self.thresh_tdiff,
-            smallest_vpfloat=constants.DBL_EPS,
-            kh_smag_e=self.kh_smag_e,
-            horizontal_start=self._edge_start_nudging,
-            horizontal_end=self._edge_end_halo,
-            vertical_start=(self._grid.num_levels - 2),
-            vertical_end=self._grid.num_levels,
-            offset_provider=self._grid.offset_providers,
-        )
-        log.debug(
-            "running stencils 11 12 (calculate_enhanced_diffusion_coefficients_for_grid_point_cold_pools): end"
-        )
+        if self.config.apply_to_temperature:
```

Comment:
So that stencil should be inside the `if` and it was not before? We never catch these kind of bugs because we allways run the same configuration. So we only ever test on branch of ifs... :-( From this point of view it is very bad that the configuration of EXCLAIM.APE is similar to MCH_CH_R04B09_DSL. We had a discussion on this yesterday. We really should come up with a list of configurations that we want to support and then test all of them and delete what we don't want and test. @anuragdipankar @lxavier @muellch

Reply:
Yes, those stencils are inside the `if` statement. The temperature diffusion should be turned off in the standard Jablonowski Williamson test.